### ISSUE TITLE
[🐛] launchImageLibrary does not work on Android.

### ISSUE THREAD

**Issue by ntalakhadze** [react-native-image-picker/react-native-image-picker#2108]
I'm using newest version 5.3.1
When I tap launchImageLibrary on Android it throws  Possible Unhandled Promise Rejection (id: 0):
TypeError: Cannot read property 'launchImageLibrary' of null.
It does not even open up the gallery.

- Image Picker version: 5.3.1
- React Native version: 0.64
- Platform: Android

buildscript {
    ext {
        buildToolsVersion = "29.0.3"
        minSdkVersion = 21
        compileSdkVersion = 31
        targetSdkVersion = 31
        ndkVersion = "20.1.5948944"
        facebookSdkVersion = "13.1.0"
    }

**Comment by waska14** [comment]
I came here to mention the same issue.. 
Do we have any suggestions or workaround here?

**Comment by viceplayer** [comment]
Hi, I have the same issue. Any chance creator could help us?

**Comment by Johan-dutoit** [comment]
You'll need to target SDK version 33 to use the latest version.  It's worth trying the example app and using that as a guide on setup.

**Comment by ntalakhadze** [comment]
Hi @Johan-dutoit . I tried that and it still does not work

**Comment by viceplayer** [comment]
Same issue here... @Johan-dutoit  not working

**Comment by Johan-dutoit** [comment]
Have you tried running the example app, that works for me and in other projects.

**Comment by ntalakhadze** [comment]
Yes I have tried it, but in Android I get this error: TypeError: Cannot read property 'launchImageLibrary' of null, when I call ImagePicker.launchImageLibrary(options, setResponse);
I switched to target Sdk 33 as well.

`
  const [response, setResponse] = useState(null);
  const options = {
    selectionLimit: 1,
    mediaType: 'video',
    formatAsMp4: true,
    includeExtra,
  };
  const onButtonPress = React.useCallback((type, options) => {
    if (type === 'capture') {
      ImagePicker.launchCamera(options, setResponse);
    } else {
      ImagePicker.launchImageLibrary(options, setResponse);
    }
    console.log(response);
  }, []);


 <TouchableOpacity onPress={() => onButtonPress('library', options)}>
          <View style={styles.uploadStyle}>
            <Image
              style={styles.uploadImage}
              source={require('../../assets/gallery.png')}
            />
            <Text style={styles.title}>Upload video from Gallery</Text>
          </View>
        </TouchableOpacity>
`

**Comment by tbesi** [comment]
Had the same issue today for both launchImageLibrary and launchCamera. Changing targetSdkVersion to 33 worked for me.

React Native Version: 0.68.5
buildToolsVersion = "31.0.0"
minSdkVersion = 21
compileSdkVersion = 31
targetSdkVersion = 33
ndkVersion = "24.0.8215888"

**Comment by kidasovmercury** [comment]
Doesn't work for me as well

**Comment by whyccup** [comment]
The issue seems to be caused by not providing the mediaType parameter when calling launchImageLibrary(). To resolve this issue, you should ensure that the mediaType parameter is always passed when invoking the function.

```js
const result = await launchImageLibrary({
  mediaType: 'photo' // or 'video' or 'mixed' depending on your use case
});
```

**Comment by trungtruc91** [comment]
same issue

**Comment by foresthu2006** [comment]
Same issue for me, tried older versions (up to 3.8.1) and they gave the same error

**Comment by robertgagnon726** [comment]
I'm having the same issue. I've tried the following and nothing has worked yet:

- Reinstalling pods
- Importing as `import * as ImagePicker from 'react-native-image-picker'` 
- importing as `const ImagePicker = require('react-native-image-picker')`

I have the mediaType prop set the image.

In the error stack into looks like it's an issue on the native layer

<img width="911" alt="Screenshot 2023-04-22 at 5 30 15 PM" src="https://user-images.githubusercontent.com/24251520/233807480-fe494d24-2c91-49df-a04c-78d244268c5b.png">

**Comment by dekemega** [comment]
Same issue for me

**Comment by Feran12** [comment]
Same issue for me but on IOS

**Comment by robertgagnon726** [comment]
I solved this by quitting the simulator and rebuilding with `react-native run-ios`

**Comment by ilyavf** [comment]
Same for me with  `targetSdkVersion = 31`. Is it possible to fix for this targetSdkVersion?

**Comment by ilyavf** [comment]
Similar issue reported here: https://github.com/react-native-image-picker/react-native-image-picker/issues/2096

**Comment by johancberg** [comment]
I solved this problem by importing the function only instead of the entire class. That is changing the import from...

`import ImagePicker from 'react-native-image-picker';`

... to the following

`import {launchImageLibrary} from 'react-native-image-picker';`

This made the mentioned error disappear and the gallery would open up for me at least.

**Comment by fragilebanana16** [comment]
if you are using expo, try https://docs.expo.dev/versions/latest/sdk/imagepicker/?redirected
i tried react-native-document-picker, react-native-image-picker, both of them wont work, use expo-image-picker instead

**Comment by tomchemitrov** [comment]
I solved this issue with installing the library with **yarn**.
**yarn add react-native-image-picker**

Previously, it was installed with npm.
Hope this helps.

**Comment by Martyneris** [comment]
any updates on this? 
The issue still persists for me on react-native: 0.64.2,  react-native-image-picker: 7.0.0 , targetSdkVersion 33

**Comment by brunxav** [comment]
I resolved this by changing the import method.

From: import ImagePicker from 'react-native-image-picker';

To: import * as ImagePicker from 'react-native-image-picker';

**Comment by BilgehanH** [comment]
Same issue for me

**Comment by budiboysupertramp** [comment]
like this import * as ImagePicker from 'react-native-image-picker';
